tests/test_public.py

```diff
@@ -0,0 +1,34 @@
+from os import path, listdir
+import subprocess
+import shutil
+
+config_path = path.join("tests", "test_config.yaml")
+files = ["test1.cpp", "test2.cpp", "test3.cpp"]
```

Comment:
After that, you can use `listdir` to find all cpp files in the input folder. Also, I think it can be done inside the test function because not all tests require all files.